src/acl_usm.cpp

```diff
@@ -606,6 +612,14 @@ CL_API_ENTRY cl_int CL_API_CALL clGetMemAllocInfoINTEL(
     }
   } break;
 
+  case CL_MEM_ALLOC_BUFFER_LOCATION_INTEL: {
+    if (usm_alloc) {
+      RESULT_UINT(usm_alloc->mem->mem_id);
+    } else {
+      RESULT_UINT(CL_MEM_TYPE_UNKNOWN_INTEL);
```

Comment:
Where is this return value specified? The [cl_intel_mem_alloc_buffer_location specification](https://github.com/KhronosGroup/OpenCL-Docs/blob/591eb712d8d9a7d47d1d48437ae84521a3c03121/extensions/cl_intel_mem_alloc_buffer_location.asciidoc#modifications-to-the-opencl-api-specification) mentions this:
> Returns buffer location for the Unified Shared Memory allocation.
>
> Return the ID of default global memory if no buffer location property were specified.

Reply:
I think the spec is meant to say when no property is passed into clDeviceMemAlloc, then default mem is returned, but in this case if the usm pointer itself is null, the behavior is not defined in the spec. I will just make it return 0 (I am surprised that this function does not error out when usm pointer is null).